data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/node/GetSyncingTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.beaconrestapi.handlers.v1.node;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static tech.pegasys.teku.beaconrestapi.CacheControlUtils.CACHE_NONE;
+
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import tech.pegasys.teku.api.SyncDataProvider;
+import tech.pegasys.teku.api.response.v1.node.SyncingResponse;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.sync.SyncService;
+
+public class GetSyncingTest {
+  private final JsonProvider jsonProvider = new JsonProvider();
+  private final SyncService syncService = mock(SyncService.class);
+  private final SyncDataProvider syncDataProvider = new SyncDataProvider(syncService);
+  private final Context context = mock(Context.class);
+
+  private final ArgumentCaptor<String> stringArgs = ArgumentCaptor.forClass(String.class);
+
+  @Test
+  public void shouldGetSyncingStatusSyncing() throws Exception {
+    GetSyncing handler = new GetSyncing(syncDataProvider, jsonProvider);
+    when(syncService.getSyncStatus()).thenReturn(getSyncStatus(true, 1, 7, 10));
+    handler.handle(context);
+    verify(context).result(stringArgs.capture());
+    verify(context).header(Header.CACHE_CONTROL, CACHE_NONE);
```

Comment:
Since we're going to write a bunch of tests like this, I wonder if we should extract a helper library that can manage the mock context for us.  So instead of having to manually create the stringArgs.capture etc, we can have something like:

```
final ContextTestHelper contextHelper = new ContextTestHelper();

handler.handle(contextHelper.getContext());

assertThat(contextHelper.getResult(SyncingResponse.class)).isEqualTo(expected);
contextHelper.assertNotCacheable();
```

It would just create a mock `Context` internally and do basically the same thing you are, but we'd get much more readable tests with less duplicate boilerplate.

Reply:
agreed, i'll work on that with the next set...